src/DBFTables.jl

```diff
@@ -452,8 +452,9 @@ function Base.getproperty(dbf::Table, name::Symbol)
     nrow = header.records
     @inbounds field = getfields(dbf)[col]
     str = getstrings(dbf)
-    @inbounds colarr = [julia_value(field, str[col, i]) for i = 1:nrow]
-    return colarr
+    ft = field.type
+    fv = Val{field.dbf_type}()
+    return @inbounds Union{field.type, Missing}[julia_value(ft, fv, str[col, i]) for i = 1:nrow]
```

Comment:
Ok if this performs fine we can just clean it up a bit for reading later:
```suggestion
    FT = field.type
    FV = Val{field.dbf_type}()
    return @inbounds Union{FT, Missing}[julia_value(FT, FV, str[col, i]) for i = 1:nrow]
```

Its also kind of confusing that one type is on its own and one is inside `Val`